dxr/app.py

```diff
@@ -165,6 +167,29 @@ def _tree_tuples(trees, tree, query_text, is_case_sensitive):
              description)
             for t, description in trees.iteritems()]
 
+@dxr_blueprint.route('/<tree>/raw/<path:path>')
+def raw(tree, path):
+    """Send raw data at path from tree, for binary things like images."""
+    query = {
+        'filter': {
+            'term': {
+                'path': path
+            }
+        }
+    }
+    index = current_app.config['ES_ALIASES'][tree]
+    results = current_app.es.search(
+            query,
+            index=index,
```

Comment:
I hate to say it, but now that we've made FILE docs potentially large, we should avoid needlessly bringing the image over the wire in browse(), likely using http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-fields.html or http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-source-filtering.html.